### PR TITLE
Change default room version to 10. Implements MSC3904

### PR DESCRIPTION
**Pull request by FSG-Cat**
This PR implements MSC3904 and is not expected to be merged before MSC3904 has passed FCP. 

Changelogs might not be done completely correctly. If there are issues with them please point it out. 

### Pull Request Checklist

* [X] Pull request is based on the develop branch
* [X] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [X] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [X] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  
Signed-off-by: Catalan Lover <catalanlover@protonmail.com>

### CODE REVIEW

File: changelog.d/14111.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Change documentation to reflect that room version 10 is now default. Contributed by @FSG-Cat.
```
Comment (DMRobertson) [inline changelog.d/14111.doc:1]:
```suggestion
Mark [room version 10](https://spec.matrix.org/v1.4/rooms/v10/) as the default room version. Contributed by @FSG-Cat.
```

File: changelog.d/14111.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Make room version 10 default. Implements MSC3904. Contributed by @FSG-Cat.
```
Comment (DMRobertson) [inline changelog.d/14111.feature:1]:
```suggestion
Update the default room version to [v10](https://spec.matrix.org/v1.4/rooms/v10/) ([MSC 3904](https://github.com/matrix-org/matrix-spec-proposals/pull/3904)). Contributed by @FSG-Cat.

File: docs/usage/configuration/config_documentation.md [hunk 3]
```diff
@@ -295,7 +295,7 @@ Known room versions are listed [here](https://spec.matrix.org/latest/rooms/#comp
 For example, for room version 1, `default_room_version` should be set
 to "1".
 
-Currently defaults to "9".
+Currently defaults to "10".
```
Comment (DMRobertson) [inline docs/usage/configuration/config_documentation.md:298]:
```suggestion
Currently defaults to ["10"](https://spec.matrix.org/v1.4/rooms/v10/).

_Changed in Synapse X.Y:_ the default version room version was increased from [9](https://spec.matrix.org/v1.4/rooms/v9/) to [10](https://spec.matrix.org/v1.4/rooms/v10/).
```

**Review by DMRobertson — CHANGES_REQUESTED** [review]
Blocked on https://github.com/matrix-org/matrix-spec-proposals/pull/3904#pullrequestreview-1135375408?

**Comment by FSG-Cat** [comment]
> Blocked on [matrix-org/matrix-spec-proposals#3904 (review)](https://github.com/matrix-org/matrix-spec-proposals/pull/3904#pullrequestreview-1135375408)?

Yes this PR is dependent on MSC3904 and that MSC is dependent on Conduit getting v10 support and enough time having passed after that point.

**Comment by clokep** [comment]
@FSG-Cat Would you mind dusting this off now that the MSC is in FCP? I think we need to:

1. Merge in `develop`.
2. Update the changelog based on @DMRobertson's comment above: https://github.com/matrix-org/synapse/pull/14111#discussion_r991043401.
3. Re-request review on it (or shout here and one of us will put it back in the queue).

Thanks so much! 👍

**Comment by FSG-Cat** [comment]
Ok so i fixed the problems i think but also i noticed the Synapse docs are very inconsistent in if they refer to latest spec or a spec version. I have decided that for my PR im linking to 1.5 as its the latest release but im sure that it might be a goal for another PR to go thru the docs and see what links need to be latest like for example known room versions might want to be latest and what links should be set manually if any. 

For example the text below is from the docs and its an example of where we use Explicit versions but in other places you can find latest references. 

> Client requests that invite user(s) when [creating a
room](https://spec.matrix.org/v1.2/client-server-api/#post_matrixclientv3createroom)
will count against the `rc_invites.per_room` limit, whereas
client requests to [invite a single user to a
room](https://spec.matrix.org/v1.2/client-server-api/#post_matrixclientv3roomsroomidinvite)
will count against both the `rc_invites.per_user` and `rc_invites.per_room` limits.

**Comment by clokep** [comment]
> im sure that it might be a goal for another PR to go thru the docs and see what links need to be latest like for example known room versions might want to be latest and what links should be set manually if any.

This sounds unrelated to this PR; if you wouldn't mind filing an issue that would be great!

**Comment by FSG-Cat** [comment]
> > im sure that it might be a goal for another PR to go thru the docs and see what links need to be latest like for example known room versions might want to be latest and what links should be set manually if any.
> 
> This sounds unrelated to this PR; if you wouldn't mind filing an issue that would be great!

Oh yes i am sure its work for another PR i just wanted to write it down. And yes i will happily open an issue.

**Comment by clokep** [comment]
This seems to have failing tests? Unsure if it is from this branch or not though.

**Comment by FSG-Cat** [comment]
> This seems to have failing tests? Unsure if it is from this branch or not though.

Dendrite reported that they had an issue with Sytest failing due to it not knowing about v10. Not sure if that is related to the test failures. Like after all the only change this PR makes is making v10 default and well anyone who has run v10 default in production for the last few months like my self can tell this works.

**Comment by clokep** [comment]
Sytest PR up at matrix-org/sytest#1328, once that merges we can kick off the builds again.

**Comment by clokep** [comment]
@DMRobertson I think this will pass tests now! 👍

File: docs/usage/configuration/config_documentation.md [hunk 4]
```diff
@@ -295,7 +295,9 @@ Known room versions are listed [here](https://spec.matrix.org/latest/rooms/#comp
 For example, for room version 1, `default_room_version` should be set
 to "1".
 
-Currently defaults to "9".
+Currently defaults to ["10"](https://spec.matrix.org/v1.5/rooms/v10/).
+
+_Changed in Synapse X.Y:_ the default version room version was increased from [9](https://spec.matrix.org/v1.5/rooms/v9/) to [10](https://spec.matrix.org/v1.5/rooms/v10/).
```
Comment (DMRobertson) [inline docs/usage/configuration/config_documentation.md:300]:
```suggestion
_Changed in Synapse 1.76:_ the default version room version was increased from [9](https://spec.matrix.org/v1.5/rooms/v9/) to [10](https://spec.matrix.org/v1.5/rooms/v10/).
```

**Review by DMRobertson — APPROVED** [review]
Otherwise, I'm happy if CI is

File: docs/usage/configuration/config_documentation.md (hunk 4 above)
Comment (clokep) [reply docs/usage/configuration/config_documentation.md]: Good catch!

**Pull request merged**